backend/src/sms/middlewares/sms-transactional.middleware.ts

```diff
@@ -3,32 +3,75 @@ import { SmsTransactionalService } from '@sms/services'
 import { loggerWithLabel } from '@core/logger'
 import { TemplateError } from '@shared/templating'
 import { RateLimitError, InvalidRecipientError } from '@core/errors'
+import { SmsMessageTransactional } from '@sms/models/sms-message-transactional'
 
 const logger = loggerWithLabel(module)
 
+function convertMessageModelToResponse(message: SmsMessageTransactional) {
+  return {
+    id: message.id,
+    credentialsLabel: message.credentialsLabel,
+    recipient: message.recipient,
+    body: message.body,
+    message_id: message.messageId,
+    created_at: message.createdAt,
+    updated_at: message.updatedAt,
+  }
+}
+
+async function saveMessage(
+  req: Request,
+  _: Response,
+  next: NextFunction
+): Promise<void> {
+  const action = 'saveMessage'
+  logger.info({ message: 'Saving SMS', action })
+  const { recipient, body, label } = req.body
+
+  const smsMessageTransactional = await SmsMessageTransactional.create({
+    userId: req.session?.user?.id,
+    recipient,
+    body,
+    credentialsLabel: label,
+    messageId: null,
+    // not sure why unknown is needed to silence TS (yet other parts of the code base can just use `as Model` directly hmm)
+  } as unknown as SmsMessageTransactional)
+  req.body.smsMessageTransactionalId = smsMessageTransactional.id // for subsequent middlewares to distinguish whether this is a transactional SMS
+  next()
+}
+
 async function sendMessage(
   req: Request,
   res: Response,
   next: NextFunction
 ): Promise<void> {
-  const ACTION = 'sendMessage'
+  const action = 'sendMessage'
 
   const { credentials } = res.locals
   try {
-    const { recipient, body } = req.body
-
-    logger.info({ message: 'Sending SMS', action: ACTION })
-    await SmsTransactionalService.sendMessage({
+    const smsMessageTransactional = await SmsMessageTransactional.findByPk(
+      req.body.smsMessageTransactionalId
+    )
+    if (!smsMessageTransactional) {
+      throw new Error('Unable to find entry in sms_messages_transactional')
+    }
+    logger.info({ message: 'Sending transactional SMS (middleware)', action })
```

Comment:
This log could be a bit more useful if we replace/add on top of `action` the message object

Reply:
But I know it's legacy and we're perhaps better off refactoring all of these in one shot

Reply:
we create a separate issue to deal with this to overhaul `logger.info`
- we alr have trace id (on DataDog), so we can correlate logs across different functions, so we don't need to log message in detail
- no point logging these happy paths `logger.info` across the entire codebase

I'll create an issue after this

Reply:
created https://github.com/opengovsg/postmangovsg/issues/1902